update-format-crau/src/bin/test.rs

```diff
@@ -23,7 +29,14 @@ impl DeltaUpdateFileHeader {
     }
 }
 
-fn read_delta_update_header(f: &mut dyn Read) -> Result<DeltaUpdateFileHeader, Box<dyn Error>> {
+// SigBuffer holds a byte slice containing signature data, and its length.
+struct SigBuffer<'a> {
+    data: &'a mut [u8],
+    len: u64,
```

Comment:
slices already have inherent length - is the length of `data` different from `len` here?

Reply:
Right. I was making it over-complicated.
Will simply remove SigBuffer, and pass the data alone.